### ISSUE TITLE
MM_Grounding_DION 训练第4个epoch时，coco val 结果突然无效

### ISSUE THREAD

**Issue by lluo-Desktop** [open-mmlab/mmdetection#11459]
**Describe the bug**
1）data prepare：按照readme完成数mm_grounding_dino复现的环境、数据准备
2）training from scratch ：进行mm_grounding_dion-tiny在数据集objects365v1-odvg格式训练
3）前5个epoch 的val log：
2024/02/02 17:02:00 - mmengine - INFO - Epoch(val) [1][625/625]    coco/bbox_mAP: 0.2940  coco/bbox_mAP_50: 0.4100  coco/bbox_mAP_75: 0.3180  coco/bbox_mAP_s: 0.2020  coco/bbox_mAP_m: 0.3140  coco/bbox_mAP_l: 0.3800  data_time: 0.0028  time: 0.2048
2024/02/03 06:32:38 - mmengine - INFO - Epoch(val) [2][625/625]    coco/bbox_mAP: 0.3050  coco/bbox_mAP_50: 0.4280  coco/bbox_mAP_75: 0.3290  coco/bbox_mAP_s: 0.2070  coco/bbox_mAP_m: 0.3290  coco/bbox_mAP_l: 0.4040  data_time: 0.0024  time: 0.2051
2024/02/03 20:04:15 - mmengine - INFO - Epoch(val) [3][625/625]    coco/bbox_mAP: 0.3250  coco/bbox_mAP_50: 0.4510  coco/bbox_mAP_75: 0.3520  coco/bbox_mAP_s: 0.2200  coco/bbox_mAP_m: 0.3530  coco/bbox_mAP_l: 0.4180  data_time: 0.0023  time: 0.2040
2024/02/04 09:37:41 - mmengine - INFO - Epoch(val) [4][625/625]    coco/bbox_mAP: 0.0040  coco/bbox_mAP_50: 0.0050  coco/bbox_mAP_75: 0.0040  coco/bbox_mAP_s: 0.0030  coco/bbox_mAP_m: 0.0060  coco/bbox_mAP_l: 0.0060  data_time: 0.0023  time: 0.2041
2024/02/04 23:12:06 - mmengine - INFO - Epoch(val) [5][625/625]    coco/bbox_mAP: 0.0020  coco/bbox_mAP_50: 0.0030  coco/bbox_mAP_75: 0.0020  coco/bbox_mAP_s: 0.0020  coco/bbox_mAP_m: 0.0030  coco/bbox_mAP_l: 0.0030  data_time: 0.0022  time: 0.2038
4）对比第3个epoch结束时的loss print和第4个epoch结束时的loss print ， 发现xxx_loss_cls变的很大：
2024/02/03 19:56:51 - mmengine - INFO - Epoch(train)  [3][18950/19019]  base_lr: 4.0000e-04 lr: 4.0000e-04  eta: 15 days, 2:33:45  time: 2.5334  data_time: 0.0126  memory: 11789  grad_norm: 22.9636  loss: 11.2329  loss_cls: 0.3225  loss_bbox: 0.1203  loss_iou: 0.4270  d0.loss_cls: 0.3911  d0.loss_bbox: 0.1278  d0.loss_iou: 0.4429  d1.loss_cls: 0.3479  d1.loss_bbox: 0.1250  d1.loss_iou: 0.4389  d2.loss_cls: 0.3332  d2.loss_bbox: 0.1216  d2.loss_iou: 0.4317  d3.loss_cls: 0.3267  d3.loss_bbox: 0.1202  d3.loss_iou: 0.4261  d4.loss_cls: 0.3225  d4.loss_bbox: 0.1198  d4.loss_iou: 0.4270  enc_loss_cls: 0.3733  enc_loss_bbox: 0.1469  enc_loss_iou: 0.4847  dn_loss_cls: 0.1241  dn_loss_bbox: 0.1939  dn_loss_iou: 0.4228  d0.dn_loss_cls: 0.1809  d0.dn_loss_bbox: 0.2831  d0.dn_loss_iou: 0.5807  d1.dn_loss_cls: 0.1479  d1.dn_loss_bbox: 0.2166  d1.dn_loss_iou: 0.4578  d2.dn_loss_cls: 0.1341  d2.dn_loss_bbox: 0.1981  d2.dn_loss_iou: 0.4303  d3.dn_loss_cls: 0.1288  d3.dn_loss_bbox: 0.1939  d3.dn_loss_iou: 0.4220  d4.dn_loss_cls: 0.1247  d4.dn_loss_bbox: 0.1938  d4.dn_loss_iou: 0.4221
2024/02/03 19:58:58 - mmengine - INFO - Epoch(train)  [3][19000/19019]  base_lr: 4.0000e-04 lr: 4.0000e-04  eta: 15 days, 2:31:39  time: 2.5440  data_time: 0.0112  memory: 11412  grad_norm: 22.9334  loss: 10.2926  loss_cls: 0.2827  loss_bbox: 0.1157  loss_iou: 0.3293  d0.loss_cls: 0.3465  d0.loss_bbox: 0.1225  d0.loss_iou: 0.3411  d1.loss_cls: 0.3083  d1.loss_bbox: 0.1210  d1.loss_iou: 0.3385  d2.loss_cls: 0.2899  d2.loss_bbox: 0.1199  d2.loss_iou: 0.3344  d3.loss_cls: 0.2846  d3.loss_bbox: 0.1158  d3.loss_iou: 0.3305  d4.loss_cls: 0.2837  d4.loss_bbox: 0.1152  d4.loss_iou: 0.3287  enc_loss_cls: 0.3298  enc_loss_bbox: 0.1462  enc_loss_iou: 0.3849  dn_loss_cls: 0.1372  dn_loss_bbox: 0.2190  dn_loss_iou: 0.3976  d0.dn_loss_cls: 0.1885  d0.dn_loss_bbox: 0.3102  d0.dn_loss_iou: 0.5505  d1.dn_loss_cls: 0.1581  d1.dn_loss_bbox: 0.2428  d1.dn_loss_iou: 0.4318  d2.dn_loss_cls: 0.1463  d2.dn_loss_bbox: 0.2236  d2.dn_loss_iou: 0.4057  d3.dn_loss_cls: 0.1408  d3.dn_loss_bbox: 0.2190  d3.dn_loss_iou: 0.3978  d4.dn_loss_cls: 0.1379  d4.dn_loss_bbox: 0.2190  d4.dn_loss_iou: 0.3976
2024/02/04 09:29:44 - mmengine - INFO - Epoch(train)  [4][18950/19019]  base_lr: 4.0000e-04 lr: 4.0000e-04  eta: 14 days, 13:26:00  time: 2.5552  data_time: 0.0114  memory: 11674  grad_norm: 133.1395  loss: 18.3762  loss_cls: 0.8443  loss_bbox: 0.1459  loss_iou: 0.4744  d0.loss_cls: 0.9097  d0.loss_bbox: 0.1523  d0.loss_iou: 0.4935  d1.loss_cls: 0.8732  d1.loss_bbox: 0.1502  d1.loss_iou: 0.4861  d2.loss_cls: 0.8552  d2.loss_bbox: 0.1481  d2.loss_iou: 0.4792  d3.loss_cls: 0.8484  d3.loss_bbox: 0.1455  d3.loss_iou: 0.4738  d4.loss_cls: 0.8439  d4.loss_bbox: 0.1459  d4.loss_iou: 0.4735  enc_loss_cls: 0.9036  enc_loss_bbox: 0.1766  enc_loss_iou: 0.5439  dn_loss_cls: 0.5860  dn_loss_bbox: 0.2076  dn_loss_iou: 0.4502  d0.dn_loss_cls: 0.6081  d0.dn_loss_bbox: 0.2920  d0.dn_loss_iou: 0.6066  d1.dn_loss_cls: 0.5949  d1.dn_loss_bbox: 0.2274  d1.dn_loss_iou: 0.4869  d2.dn_loss_cls: 0.5898  d2.dn_loss_bbox: 0.2122  d2.dn_loss_iou: 0.4588  d3.dn_loss_cls: 0.5881  d3.dn_loss_bbox: 0.2076  d3.dn_loss_iou: 0.4494  d4.dn_loss_cls: 0.5862  d4.dn_loss_bbox: 0.2075  d4.dn_loss_iou: 0.4495
2024/02/04 09:31:50 - mmengine - INFO - Epoch(train)  [4][19000/19019]  base_lr: 4.0000e-04 lr: 4.0000e-04  eta: 14 days, 13:23:46  time: 2.5234  data_time: 0.0117  memory: 11252  grad_norm: 123.7338  loss: 18.8641  loss_cls: 0.8534  loss_bbox: 0.1578  loss_iou: 0.4723  d0.loss_cls: 0.9101  d0.loss_bbox: 0.1682  d0.loss_iou: 0.4859  d1.loss_cls: 0.8767  d1.loss_bbox: 0.1636  d1.loss_iou: 0.4806  d2.loss_cls: 0.8648  d2.loss_bbox: 0.1607  d2.loss_iou: 0.4781  d3.loss_cls: 0.8580  d3.loss_bbox: 0.1571  d3.loss_iou: 0.4715  d4.loss_cls: 0.8541  d4.loss_bbox: 0.1575  d4.loss_iou: 0.4713  enc_loss_cls: 0.9060  enc_loss_bbox: 0.1925  enc_loss_iou: 0.5369  dn_loss_cls: 0.6231  dn_loss_bbox: 0.2191  dn_loss_iou: 0.4611  d0.dn_loss_cls: 0.6468  d0.dn_loss_bbox: 0.3122  d0.dn_loss_iou: 0.6243  d1.dn_loss_cls: 0.6313  d1.dn_loss_bbox: 0.2421  d1.dn_loss_iou: 0.5008  d2.dn_loss_cls: 0.6269  d2.dn_loss_bbox: 0.2229  d2.dn_loss_iou: 0.4704  d3.dn_loss_cls: 0.6244  d3.dn_loss_bbox: 0.2190  d3.dn_loss_iou: 0.4605  d4.dn_loss_cls: 0.6228  d4.dn_loss_bbox: 0.2191  d4.dn_loss_iou: 0.4604

**Reproduction**
1.What command or script did you run?
./tools/dist_train.sh ./configs/mm_grounding_dino/grounding_dino_swin-t_pretrain_obj365.py 8
2. I did not make any modifications on the code or config? 
3. What dataset did you use? - objects365_odvg

**Environment**
Python: 3.8.18 | packaged by conda-forge | (default, Dec 23 2023, 17:21:28) [GCC 12.3.0]
CUDA available: True
numpy_random_seed: 2147483648
GPU 0: Tesla V100-SXM2-32GB
CUDA_HOME: /usr/local/cuda
NVCC: Cuda compilation tools, release 11.3, V11.3.109
GCC: gcc (Ubuntu 7.5.0-3ubuntu1~18.04) 7.5.0
PyTorch: 1.12.0+cu113
PyTorch compiling details: PyTorch built with:
  - GCC 9.3
  - C++ Version: 201402
  - Intel(R) Math Kernel Library Version 2020.0.0 Product Build 20191122 for Intel(R) 64 architecture applications
  - Intel(R) MKL-DNN v2.6.0 (Git Hash 52b5f107dd9cf10910aaa19cb47f3abf9b349815)
  - OpenMP 201511 (a.k.a. OpenMP 4.5)
  - LAPACK is enabled (usually provided by MKL)
  - NNPACK is enabled
  - CPU capability usage: AVX2
  - CUDA Runtime 11.3
  - NVCC architecture flags: -gencode;arch=compute_37,code=sm_37;-gencode;arch=compute_50,code=sm_50;-gencode;arch=compute_60,code=sm_60;-gencode;arch=compute_70,code=sm_70;-gencode;arch=compute_75,code=sm_75;-gencode;arch=compute_80,code=sm_80;-gencode;arch=compute_86,code=sm_86
  - CuDNN 8.2
    - Built with CuDNN 8.3.2
  - Magma 2.5.2
  - Build settings: BLAS_INFO=mkl, BUILD_TYPE=Release, CUDA_VERSION=11.3, CUDNN_VERSION=8.3.2, CXX_COMPILER=/opt/rh/devtoolset-9/root/usr/bin/c++, CXX_FLAGS= -Wno-deprecated -fvisibility-inlines-hidden -DUSE_PTHREADPOOL -fopenmp -DNDEBUG -DUSE_KINETO -DUSE_FBGEMM -DUSE_QNNPACK -DUSE_PYTORCH_QNNPACK -DUSE_XNNPACK -DSYMBOLICATE_MOBILE_DEBUG_HANDLE -DEDGE_PROFILER_USE_KINETO -O2 -fPIC -Wno-narrowing -Wall -Wextra -Werror=return-type -Wno-missing-field-initializers -Wno-type-limits -Wno-array-bounds -Wno-unknown-pragmas -Wno-unused-parameter -Wno-unused-function -Wno-unused-result -Wno-unused-local-typedefs -Wno-strict-overflow -Wno-strict-aliasing -Wno-error=deprecated-declarations -Wno-stringop-overflow -Wno-psabi -Wno-error=pedantic -Wno-error=redundant-decls -Wno-error=old-style-cast -fdiagnostics-color=always -faligned-new -Wno-unused-but-set-variable -Wno-maybe-uninitialized -fno-math-errno -fno-trapping-math -Werror=format -Werror=cast-function-type -Wno-stringop-overflow, LAPACK_INFO=mkl, PERF_WITH_AVX=1, PERF_WITH_AVX2=1, PERF_WITH_AVX512=1, TORCH_VERSION=1.12.0, USE_CUDA=ON, USE_CUDNN=ON, USE_EXCEPTION_PTR=1, USE_GFLAGS=OFF, USE_GLOG=OFF, USE_MKL=ON, USE_MKLDNN=OFF, USE_MPI=OFF, USE_NCCL=ON, USE_NNPACK=ON, USE_OPENMP=ON, USE_ROCM=OFF, 

TorchVision: 0.13.0+cu113
OpenCV: 4.8.1
MMEngine: 0.10.1
MMDetection: 3.3.0+

请问有遇到过相关的loss突变，val失效的问题吗？是否有可以参考的调整建议？

**Comment by hhaAndroid** [comment]
这种情况我跑了很多次，没有出现过耶。我们是 32 卡训练的，如果你不是一样的 bs， 建议可以调整下学习率。如果觉得不需要可以从 epoch 3 resume 训练可能就好了

**Comment by lluo-Desktop** [comment]
> 这种情况我跑了很多次，没有出现过耶。我们是 32 卡训练的，如果你不是一样的 bs， 建议可以调整下学习率。如果觉得不需要可以从 epoch 3 resume 训练可能就好了

参数设置基本是参照默认的grounding_dino_swin-t_pretrain_obj365.py跑的，看上去似乎存在的差异在于卡的数量：
我使用的8gpus x 4 bs , usage中提到的使用的32gpus x 4 bs。
不过我看config中设置了auto_scale_lr = dict(base_batch_size=64) 我就没有手动调整lr。
所以你的建议是如果是用8gpus时，手动调节降低lr0.0004 -> 0.0001？

**Comment by hhaAndroid** [comment]
对，学习率设小一点可能性能会更好。

**Comment by hhaAndroid** [comment]
@lluo-Desktop 我们今天发布了 swin-b 和 swin-l 权重，如果有兴趣可以看看 https://github.com/open-mmlab/mmdetection/pull/11458

**Comment by lluo-Desktop** [comment]
@hhaAndroid 好的哈，非常棒的工作！
我调整一下lr，如果有好的结果再告知你哈！

**Comment by hhaAndroid** [comment]
@lluo-Desktop 可以基于 dev-3.x 重新跑一下，dev-3.x 里面修复了一个数据增强 bug

**Comment by lluo-Desktop** [comment]
@hhaAndroid 
反馈一下：我调整学习率、更新了data aug的 fix后可以run 24epoch并且得到不错的zero-shot结果。

另外有个问题：使用image_demo去做inference，在选择“oid_v6” classes时就没有正确输出了。
我还没仔细看代码，是不是language model对max_tokens有限制？如果我想在自己的1k+ classes上做测试，有什么建议吗？

**Comment by lluo-Desktop** [comment]
好吧 设置一下chunked-size

**Comment by jinyizhang184** [comment]
@hhaAndroid 我出现了同样的问题。关于学习率，config文件里是auto_scale_lr = dict(base_batch_size=64) ，usage中提到的使用的32gpus x 4 bs，请问以哪个为准。我是16*4的训练的所以学习率应该没有auto_scale，但是在18个epoch后coco指标就从40+到0.4了
